File: businesslib/src/main/java/com/wujia/businesslib/dialog/LoadingProgressDialog.java
package com.wujia.businesslib.dialog;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import com.wujia.lib.uikit.R;


public class LoadingProgressDialog extends Dialog {
    private TextView tvTitle;
    String title = "下载中,请勿离开...";

    public LoadingProgressDialog(@NonNull Context context) {
        super(context, R.style.dialog);
        init();
    }

    public LoadingProgressDialog(@NonNull Context context, int themeResId) {
        super(context, themeResId);
    }

    private void init() {
        setContentView(R.layout.layout_loading_dialog);
        setCanceledOnTouchOutside(false);
        tvTitle = findViewById(R.id.tv_title);
        tvTitle.setText(title);
        tvTitle.setVisibility(View.VISIBLE);
    }

    public void updateProgress(int per) {
        tvTitle.setText(title + per + "%");
    }

    public void setTvTitle(String title) {
        tvTitle.setText(title);
    }

}

File: businesslib/src/main/java/com/wujia/businesslib/dialog/LoadingProgressDialog.kt
package com.wujia.businesslib.dialog

import android.app.Dialog
import android.content.Context
import android.view.View
import android.widget.TextView

import com.wujia.lib.uikit.R


class LoadingProgressDialog : Dialog {
    private var tvTitle: TextView? = null
    internal var title = "下载中,请勿离开..."

    constructor(context: Context) : super(context, R.style.dialog) {
        init()
    }

    constructor(context: Context, themeResId: Int) : super(context, themeResId) {}

    private fun init() {
        setContentView(R.layout.layout_loading_dialog)
        setCanceledOnTouchOutside(false)
        tvTitle = findViewById(R.id.tv_title)
        tvTitle!!.text = title
        tvTitle!!.visibility = View.VISIBLE
    }

    fun updateProgress(per: Int) {
        tvTitle!!.text = "$title$per%"
    }

    fun setTvTitle(title: String) {
        tvTitle!!.text = title
    }

}
